packages/safe-apps-sdk/src/types/sdk.ts

```diff
@@ -32,6 +32,19 @@ export type SignMessageParams = {
   message: string;
 };
 
+interface Type {
+  name: string;
+  type: string;
+}
+
+export type TypedMessageTypes = { [key: string]: Type[] };
+
+export type SignTypedMessageParams = {
+  domain: any;
```

Comment:
Should this be a string?

Reply:
To my best knowledge, `domain` can be anything. For example in https://medium.com/metamask/eip712-is-coming-what-to-expect-and-how-to-use-it-bb92fd1a7a26:

```
const domainData = {
    name: "My amazing dApp",
    version: "2",
    chainId: parseInt(web3.version.network, 10),
    verifyingContract: "0x1C56346CD2A2Bf3202F771f50d3D14a367B48070",
    salt: "0xf2d857f4a3edcb9b78b4d503bfe733db1e3f6cdc2b7971ee739626c97e86a558"
};
```

Reply:
Pretty sure you can't pass just any JS value. E.g. surely not HTML elements or functions?
How about `Record<any, any> | Record<any, any>[]`?

Reply:
Done, changed the type of `domain` from `any` to `Record<string, any>` (I think `domain` can't contain array either)